updated black version to 20.8b1

zyxue opened the pull request:
updated and reran black on `python/seldon_core` and `python/tests`

discussed on https://github.com/SeldonIO/seldon-core/pull/2713#issuecomment-735827039

Should black be run on all the pb2 files too like `seldon_core/proto/prediction_pb2.py`?

seldondev commented:
Hi @zyxue. Thanks for your PR.

I'm waiting for a [SeldonIO](https://github.com/orgs/SeldonIO/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/SeldonIO/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

seldondev commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **adriangonz**
You can assign the PR to them by writing `/assign @adriangonz` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["adriangonz"]} -->

RafalSkolasinski commented on python/Makefile:
```diff
@@ -95,7 +95,7 @@ push_conda:
 
 setup_linter:
-	pip install black==19.3b0
+	pip install black==black 20.8b1
```
Is this a typo?

RafalSkolasinski replied:
This should also be updated in `requirements.txt`

zyxue replied:
thanks for the catch. updated

RafalSkolasinski commented:
Nice one!

> Should black be run on all the pb2 files too like seldon_core/proto/prediction_pb2.py?

As these are auto-generated by protos compiler I don't think they should be covered by black.

seldondev commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **rafalskolasinski**
You can assign the PR to them by writing `/assign @rafalskolasinski` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["rafalskolasinski"]} -->

zyxue commented:
> Nice one!
> 
> > Should black be run on all the pb2 files too like seldon_core/proto/prediction_pb2.py?
> 
> As these are auto-generated by protos compiler I don't think they should be covered by black.

I see. I used to format them for better readability, and also because black won't change the logic in anyway. But not blacking them sounds good too. 

I've addressed your comments.

RafalSkolasinski commented:
/ok-to-test

RafalSkolasinski commented:
/test this

seldondev commented:
Wed Dec 16 16:39:25 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2769 --build=2

seldondev commented:
Wed Dec 16 16:39:36 UTC 2020
The logs for [pr-build] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2769 --build=3

seldondev commented:
Wed Dec 16 16:39:35 UTC 2020
The logs for [integration] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2769 --build=4

seldondev commented:
Wed Dec 16 16:39:37 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2769 --build=1

seldondev commented:
@zyxue: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | d718af1e378b048d0d4fe457e7f95de609ab5251 | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

RafalSkolasinski commented:
/retest

seldondev commented:
Wed Dec 16 17:35:25 UTC 2020
The logs for [integration] [5] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/5.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2769 --build=5

seldondev commented:
@zyxue: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | d718af1e378b048d0d4fe457e7f95de609ab5251 | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

zyxue commented:
when can I see the log of the failed integration test?

RafalSkolasinski commented:
there's a link in `seldondev` bot's message, they are available after the pipeline run

zyxue commented:
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/5.log
```
=================================== FAILURES ===================================
__________________________ TestPrepack.test_tfserving __________________________
[gw1] linux -- Python 3.6.10 /usr/local/bin/python

self = <test_prepackaged_servers.TestPrepack object at 0x7f28bc4376a0>
namespace = 'test-tfserving'

    def test_tfserving(self, namespace):
        spec = "../../servers/tfserving/samples/mnist_rest.yaml"
        retry_run(f"kubectl apply -f {spec}  -n {namespace}")
        wait_for_status("tfserving", namespace)
>       wait_for_rollout("tfserving", namespace)

test_prepackaged_servers.py:97: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

sdep_name = 'tfserving', namespace = 'test-tfserving', attempts = 50, sleep = 5
expected_deployments = 1

    def wait_for_rollout(
        sdep_name, namespace, attempts=50, sleep=5, expected_deployments=1
    ):
        deployment_names = []
        for _ in range(attempts):
            deployment_names = get_deployment_names(sdep_name, namespace)
            deployments = len(deployment_names)
    
            if deployments == expected_deployments:
                break
            time.sleep(sleep)
    
        error_msg = (
            f"Expected {expected_deployments} deployment(s) but got {len(deployment_names)}"
        )
        assert len(deployment_names) == expected_deployments, error_msg
    
        for deployment_name in deployment_names:
            logging.info(f"Waiting for deployment {deployment_name}")
            for _ in range(attempts):
                ret = run(
                    f"kubectl rollout status -n {namespace} deploy/{deployment_name}",
                    shell=True,
                )
                if ret.returncode == 0:
                    logging.info(f"Successfully waited for deployment {deployment_name}")
                    break
                logging.warning(
                    f"Unsuccessful wait command but retrying for {deployment_name}"
                )
                time.sleep(sleep)
            assert (
                ret.returncode == 0
>           ), f"Wait for rollout of {deployment_name} failed: non-zero return code"
E           AssertionError: Wait for rollout of tfserving-default-0-mnist-model failed: non-zero return code

seldon_e2e_utils.py:158: AssertionError
```
this seems have nothing to do with the chage here...

cliveseldon commented:
Can you rebase as this failure is fixed

zyxue commented:
rebased. Is test triggered automatically?

seldondev commented:
Thu Dec 17 16:55:27 UTC 2020
The logs for [pr-build] [6] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/6.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2769 --build=6

seldondev commented:
Thu Dec 17 16:55:32 UTC 2020
The logs for [lint] [7] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/7.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2769 --build=7

RafalSkolasinski commented:
/test integration

seldondev commented:
Thu Dec 17 17:04:59 UTC 2020
The logs for [integration] [8] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/8.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2769 --build=8

zyxue commented:
wondering if there is anything else that needs update for this PR?

RafalSkolasinski reviewed (approved):
Fantastic job!
/lgtm
/approve

RafalSkolasinski commented:
/approve

RafalSkolasinski commented:
/lgtm

seldondev commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[RafalSkolasinski](https://github.com/SeldonIO/seldon-core/pull/2769#issuecomment-753933489 "LGTM")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com#issuecomment-753933451/SeldonIO/seldon-core/blob/master/OWNERS)~~ [RafalSkolasinski]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

seldondev commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[RafalSkolasinski](https://github.com/SeldonIO/seldon-core/pull/2769#issuecomment-753933489 "LGTM")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com#issuecomment-753933489/SeldonIO/seldon-core/blob/master/OWNERS)~~ [RafalSkolasinski]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

seldondev commented:
Mon Jan  4 11:56:48 UTC 2021
The logs for [pr-build] [9] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/9.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2769 --build=9

seldondev commented:
Mon Jan  4 11:56:50 UTC 2021
The logs for [lint] [10] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/10.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2769 --build=10
